### ISSUE TITLE
S3mock-junit5 extensions not working with spring-context-indexer enabled

### ISSUE THREAD

**Issue by ilopezv** [adobe/S3Mock#751]
When an application has configured spring-context-indexer annotationProcessor in order to generate [spring indexs](https://docs.spring.io/spring-framework/docs/current/reference/html/core.html#beans-scanning-index), S3mock-junit5 extensions fails during unit test because beans contained in the artifact is not being found because it hasn´t got _META-INF/spring.components_ in it.

You can reproduce this case using example test found at [S3MockExtensionDeclarativeTest](https://github.com/adobe/S3Mock/blob/main/testsupport/junit5/src/test/java/com/adobe/testing/s3mock/junit5/sdk2/S3MockExtensionDeclarativeTest.java) or [S3MockExtensionProgrammaticTest](https://github.com/adobe/S3Mock/blob/main/testsupport/junit5/src/test/java/com/adobe/testing/s3mock/junit5/sdk2/S3MockExtensionProgrammaticTest.java) and using a simple project generated with spring initializer.
Attached [demo.zip](https://github.com/adobe/S3Mock/files/9331311/demo.zip) contains a simplest application with this configuration as described above 
 

- **Version affected**: all s3mock-junit5 including 2.4.16

- **SB version**: 2.7.2 
- **Possible solution**: Include spring-context-indexer as maven-compiler-annotation-processor in order to generate _META-INF/spring.components_ . If an application don´t use spring-indexer, component scan will work as usal, in the other hand if an application use spring-indexer all works fine because artifact has _META-INF/spring.components_ in it
- **Error obtained**: 
```
***************************
APPLICATION FAILED TO START
***************************

Description:

Field kmsKeyStore in com.adobe.testing.s3mock.S3MockApplication required a bean of type 'com.adobe.testing.s3mock.store.KmsKeyStore' that could not be found.

The injection point has the following annotations:
        - @org.springframework.beans.factory.annotation.Autowired(required=true)


Action:

Consider defining a bean of type 'com.adobe.testing.s3mock.store.KmsKeyStore' in your configuration.
```

**Comment by afranken** [comment]
@ilopezv thanks for this issue, I'll add spring indixes to the S3Mock since it does not seem to have adverse effects if somebody is not using them, and it may speed up the start of S3Mock itself as well.

Other than that, I would like to point out that when you're directly using the S3Mock in a unit test, both the libraries of your application and the libraries of the S3Mock are mixed together, and there are all kinds of problems with this approach.

I recommend that you look into using S3Mock as a Docker container to avoid these problems. 
Have a look at #418 for a longer explanation.

**Comment by ilopezv** [comment]
when is the release planned?

**Comment by afranken** [comment]
@ilopezv I was planning to release this week, but I am currently in the process of refactoring the whole application to better accommodate the various open feature requests.
This is taking longer than expected.

I have merged several 1000+ lines PRs already and I'll need a few more before I can release again, sorry.
Hopefully I'll be done by next week :(

**Comment by afranken** [comment]
@ilopezv I finished the refactoring, this was quite a big one... 
Now, with smaller classes and less duplicated code, it should be easier to implement the missing features like versions, ACLs, additional metadata, ...

I just released 2.5.0 for your convenience. :)

**Comment by ilopezv** [comment]
Thanks a lot!!! It seems to be a big release. 

Great work!!

**Comment by afranken** [comment]
@ilopezv I have bad news - this is actually a breaking change for _any application using Spring Boot_.

I will have to revert this change and make a new release, sorry.
Reading through the documentation the first time, I did not notice that if even _one_ library is found on the classpath including a Component Index, Spring will ignore all other configuration in the service, or other libraries.
For this to work, literally every library you depend on (that provides Spring services) has to provide those indexes.

**Comment by afranken** [comment]
this was discovered by another customer (where the application did break after updating) in #782

**Comment by afranken** [comment]
will close this as "won't fix", as I'm not able to solve this for you without breaking other people's integrations, sorry. :(

**Comment by ilopezv** [comment]
Sorry, i'm on vacations without computer so i can't review the Code.

As simple solution, could you expose the configuration classes as public so people with indexer could use @import / @contextconfiguration directly on them?

**Comment by afranken** [comment]
yeah, that is possible. :)

**Comment by afranken** [comment]
released 2.5.4 earlier today, now all `@Configuration` / `@ConfigurationProperties` classes are public.

Have a nice vacation!
